modules/core/src/main/scala/lucuma/odb/api/repo/TopLevelRepo.scala

```diff
@@ -171,24 +170,27 @@ abstract class TopLevelRepoBase[F[_]: Monad, I: Gid, T: TopLevelModel[I, ?]](
     val lens: Lens[Tables, Option[U]] =
       Lens[Tables, Option[U]](lensT.get(_).flatMap(f.unapply))(ou => lensT.set(ou))
 
-    val doUpdate: F[U] =
+    val doUpdate: F[Either[U, U]] =
       tablesRef.modify { oldTables =>
 
         val item   = lens.get(oldTables).toValidNec(InputError.missingReference("id", Gid[I].show(id)))
         val errors = NonEmptyChain.fromSeq(checks(oldTables))
         val result = (item, editor, errors.toInvalid(())).mapN { (oldU, state, _) =>
-          state.runS(oldU).value
+          val newU = state.runS(oldU).value
+          Either.cond(oldU =!= newU, newU, oldU) // Right => updated, Left => no update
         }
 
-        val tables = result.fold(_ => oldTables, { newU => lens.set(Some(newU))(oldTables) })
+        val tables = result.fold(_ => oldTables, {
+          _.fold(_ => oldTables, newU => lens.set(Some(newU))(oldTables))
+        })
 
         (tables, result)
       }.flatMap(_.liftTo[F])
 
     for {
-      u <- doUpdate
-      _ <- eventService.publish(edited(Event.EditType.Updated, u))
-    } yield u
+      e <- doUpdate
+      _ <- e.fold(_ => M.unit, u => eventService.publish(edited(Event.EditType.Updated, u)))
```

Comment:
I think this can be done with something like `eventService....whenM(e.isDefined)`. I don't know if it really helps

Reply:
What would this look like exactly?  `e` is an `Either[U, U]` btw, basically using left vs. right to record unchanged vs. changed along with the correct value of the `U` object.

Reply:
Oh, I thought it was a `Monad`, please disregard